### PR TITLE
Adding reference document for BigDecimal and Decimal128

### PR DESCRIPTION
**Pull request by caiolima**
This PR is adding initial version of reference documentation for Decimal128 and BigDecimal current API design.

Cc @littledan

### CODE REVIEW

File: bigdecimal-reference.md [hunk 1]
```diff
@@ -0,0 +1,568 @@
...
+  object described on [BigDecimal.round](#bigdecimalroundvalue--options). If it's not given, no rounding
+  operation will be applied, and the exact result will be returned. If the result can't be represented (due to a
+  non-terminating decimal expansion), it throws `TypeError`.
+
+Different from other arithmetic operations on `BigDecimal` constructors, we require `options` for division
+because this is the only operation where some results can't be represented as a `BigDecimal` value (e.g. when
+we divide 1m by 3m) if we don't round. With the requirement to describe how we should round the result, it's
+then possible to return a correct result for any given input.
+
+### BigDecimal.reminder(lhs, rhs [, options])
```
Comment (littledan) [inline bigdecimal-reference.md:421]: remainder (or we could call this "mod" though these operations are arguably different)

File: decimal128-reference.md [hunk 2]
```diff
@@ -0,0 +1,589 @@
...
+```js
+let o = {};
+o[2.45m] = "decimal";
+console.log(o["2.45"]); // prints "decimal"
+console.log(o[2.45m]); // prints "decimal"
+```
+
+## TypedArrays
+
+TODO
```
Comment (littledan) [inline decimal128-reference.md:583]: Let's remember to come back to this.

**Review by littledan — APPROVED** [review]
LGTM. I guess we will also want to explain the "BigDecimal + / with fixed rounding" alternative.

**Comment by caiolima** [comment]
Thank you very much for the reviews!

**Pull request merged**